capsules/extra/src/ieee802154/device.rs

```diff
@@ -135,3 +137,13 @@ pub trait RxClient {
     /// - `data_len`: Length of the data payload
     fn receive<'a>(&self, buf: &'a [u8], header: Header<'a>, data_offset: usize, data_len: usize);
 }
+
+pub trait RawRxClient {
```

Comment:
A comment here explaining that the fields here will match the explanation of `RxClient` would be nice. Also the comment should explain what differs from a normal `RxClient` (i.e., what "raw" actually means here)

Reply:
@brghena I just pushed changes adding these comments. (good catch on the missing documentation)

While I was adding this, I also softened the language to "potentially" including extra bytes for the PHY layer in the `buf` argument comment. This is to reflect https://github.com/tock/tock/pull/3933 removing the `PSDU` offset passed to RxClients that are "higher" in the stack.